src/main/java/org/apache/commons/io/FileUtils.java

```diff
@@ -2393,9 +2393,17 @@ public static void moveFile(final File srcFile, final File destFile) throws IOEx
      * @since 2.9.0
      */
     public static void moveFile(final File srcFile, final File destFile, final CopyOption... copyOptions) throws IOException {
-        validateMoveParameters(srcFile, destFile);
-        requireFile(srcFile, "srcFile");
-        requireAbsent(destFile, "destFile");
+        Objects.requireNonNull(srcFile, "source");
+        Objects.requireNonNull(destFile, "destination");
+        if (!srcFile.isFile()) {
+            if (srcFile.exists()) {
```

Comment:
Alternatively this logic could move into the private requireFile method, which would then distinguish between source files that are directories and files that don't exist, and throw a different exception in each case.

Reply:
I think I'm OK with that if that means the code would be reused from different APIs and those would therefore get consistent behavior which we can more simply document.